### PR TITLE
Feature/updates from csv exporter

### PR DESCRIPTION
**Pull request by redhug1**
### What

Merge in all appropriate parts from dp-cantabular-csv-exporter to pull together initial framework.
This is a work in progress and has parts of old csv code commented out and various todo markers.

### How to review

Visual inspection.
Run tests.

(its a work in progress framework, so don't pass comments on any todo marker lines that have '!!!' in them - thanks)

### Who can review

Ideally someone with knowledge of dp-cantabular-csv-exporter

### CODE REVIEW

File: features/compose/minio.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: '3.3'
+services: 
+  minio:
+    image: 'bitnami/minio:latest'
+    ports:
+      - '9000:9000'
+    environment:
+      - MINIO_ACCESS_KEY=minio-access-key
+      - MINIO_SECRET_KEY=minio-secret-key
+      - MINIO_DEFAULT_BUCKETS=dp-cantabular-csv-exporter
```
Comment (DavidSubiros) [inline features/compose/minio.yml:10]: I'd suggest adding a new line

File: features/compose/vault.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      image: vault:latest
+      ports:
+        - 8200:8200
+      environment:
+        - VAULT_ADDR=http://0.0.0.0:8200
+        - VAULT_API_ADDR=http://0.0.0.0:8200
+        - VAULT_ADDRESS=http://0.0.0.0:8200
+      cap_add:
+        - IPC_LOCK
+      command: vault server -dev
```
Comment (DavidSubiros) [inline features/compose/vault.yml:14]: new line

File: features/compose/dp-cantabular-csv-exporter.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+version: '3.3'
```
[the remark this answers is not in the document]
Comment (redhug1) [reply features/compose/dp-cantabular-csv-exporter.yml:1]: OK.
I'll change that in my next PR (as its already a work in progress based on this PR)

File: config/config.go [hunk 4]
```diff
@@ -16,7 +15,8 @@ type Config struct {
 	KafkaAddr                    []string      `envconfig:"KAFKA_ADDR"                     json:"-"`
 	KafkaVersion                 string        `envconfig:"KAFKA_VERSION"`
 	KafkaOffsetOldest            bool          `envconfig:"KAFKA_OFFSET_OLDEST"`
-	KafkaNumWorkers              int           `envconfig:"KAFKA_NUM_WORKERS"`
+	KafkaNumWorkers              int           `envconfig:"KAFKA_NUM_WORKERS"`               //!!! it might be best to remove this to avoid out of memory issues
```
[the remark this answers is not in the document]
Comment (redhug1) [reply config/config.go:18]: OK, comment added in next PR

**Review by DavidSubiros — APPROVED** [review]

**Pull request merged**